Doc/library/pathlib.rst

```diff
@@ -946,6 +946,106 @@ call fails (for example because the path doesn't exist).
+   will only recurse into the subdirectories whose names remain in *dirnames*;
+   this can be used to prune the search, or to impose a specific order of visiting,
+   or even to inform :meth:`Path.walk` about directories the caller creates or
+   renames before it resumes :meth:`Path.walk` again. Modifying *dirnames* when
+   *top_down* is False has no effect on the behavior of :meth:`Path.walk()`, since the
+   directories in *dirnames* have already been generated by the time *dirnames*
+   is yielded to the caller.
+
+   By default, errors from :func:`os.scandir` are ignored.  If the optional
+   argument *on_error* is specified, it should be a callable; it will be
+   called with one argument, an :exc:`OSError` instance. It can handle the
+   error to continue the walk or re-raise it to stop the walk. Note that the
+   filename is available as the ``filename`` attribute of the exception object.
+
+   By default, :meth:`Path.walk` will not follow symbolic links that
+   resolve to directories.  Set *follow_symlinks* to ``True`` to visit directories
+   pointed to by symlinks (where supported).
+
+   .. note::
+
+      Be aware that setting *follow_symlinks* to ``True`` can lead to infinite
+      recursion if a link points to a parent directory of itself. :meth:`Path.walk`
+      does not keep track of the directories it has already visited.
+
+   .. note::
+      :meth:`Path.walk` assumes that the current working directory does not change while
+      walking relative paths.
+
+   .. note::
+      :meth:`Path.walk` assumes the directories it walks are not been modified during
+      execution. For example, if a directory from *dirnames* has been replaced
+      with a symlink and *follow_symlinks* = ``False``, :meth:`Path.walk` will
+      still try to descend into it. To prevent such behavior, remove directories
+      from *dirnames* as appropriate.
+
+   .. note::
+
+      Unlike :func:`os.walk`, :meth:`Path.walk` lists symlinks to directories into
+      *filenames* if *follow_symlinks* is ``False``.
+
+   This example displays the number of bytes used by all files in each directory,
+   while ignoring ``__pycache__`` directories.
+   :file:`__pycache__` subdirectory::
+
+      from pathlib import Path
+      for root, dirs, files in Path("cpython/Lib/concurrent").walk(on_error=print):
+        print(
+            root,
+            "consumes",
+            sum((root / file).stat().st_size for file in files),
+            "bytes in",
+            len(files),
+            "non-directory files"
+        )
+        if '__pycache__' in dirs:
+              dirs.remove('__pycache__')
+
+   This next example is a simple implementation of :func:`shutil.rmtree`.
+   Walking the tree bottom-up is essential as :func:`rmdir` doesn't allow
+   deleting a directory before it is empty::
+
+      # Delete everything reachable from the directory "top",
+      # assuming there are no symbolic links.
```

Comment:
What part of this algorithm assumes that there are no symbolic links? It looks like it should work to me.

Reply:
You're right